### ISSUE TITLE
request help: set ENV variable in config.yaml, then admin API call still didn't recognize it

### ISSUE THREAD

**Issue by diaosj** [apache/apisix#3017]
### Issue description
I noticed that version v2.1 supports environment variables in the configuration file: https://github.com/apache/apisix/pull/2743.

I wrote these lines below to `conf/config.yaml`:
```yaml
etcd:
  host:
    - "http://${{ETCD_HOST}}:2379"
  prefix: "/apisix"
  timeout: 30
  tls:
    verify: false
```

I run the docker image with the corresponding environment variable and config file as below:
```bash
docker run -e TZ=Asia/Shanghai -e ETCD_HOST=`ipconfig getifaddr en0` -p 9080:9080 -p 9443:9443 -v `pwd`/conf/config-default.yaml:/usr/local/apisix/conf/config-default.yaml -v `pwd`/conf/config.yaml:/usr/local/apisix/conf/config.yaml -v `pwd`/plugins:/usr/local/apisix/apisix/plugins -v `pwd`/logs/error.log:/usr/local/apisix/logs/error.log apache/apisix:2.1-centos
```

Then I call the admin API. But the response is like this: 
```json
{
    "error_msg": "invalid http host: http://${{ETCD_HOST}}:2379, err: not matched"
}
```

I searched for this message in the repository. This message looks like from here: https://github.com/api7/lua-resty-etcd/blob/master/lib/resty/etcd/v3.lua#L155
It seems like that etcd client still use "${{ETCD_HOST}}" literally when admin API is called.

### Environment

* apisix version (cmd: `apisix version`): 2.1
* OS: docker image (apache/apisix:2.1-centos)

**Comment by membphis** [comment]
you can make a try with `master` branch. We fixed this bug for days I think

**Comment by spacewander** [comment]
Fixed by https://github.com/apache/apisix/pull/2990

**Comment by diaosj** [comment]
This fix wasn't merged to branch `v2.1`. Is branch `v2.1` immutable now? For it has been released?

**Comment by spacewander** [comment]
We haven't backported to released version yet.

**Comment by diaosj** [comment]
Ok. I think I can try applying these changes to my local files. Just confirm the branch management strategy.

**Comment by diaosj** [comment]
> We haven't backported to released version yet.

Just make sure, will we backport this to the released `v2.1` version? I think this backport may be necessary, but I'm not sure if it goes against the branch strategy or plan. I can contribute to this backport if you want.

**Comment by spacewander** [comment]
> Just make sure, will we backport this to the released v2.1 version?

Even we backport it to v2.1, we still need to create a new release as v2.1.1. Currently there is no plan and resource to create/maintain the patch release. But we might change it after v2.2. There is some internal discussion among APISIX members.
